ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jersey/microprofile/restclient/ExecutorServiceWrapper.java

```diff
@@ -112,15 +111,21 @@ public void execute(Runnable command) {
     }
 
     private <T> Callable<T> wrap(Callable<T> task) {
+        List<AsyncInvocationInterceptor> asyncInvocationInterceptors = asyncInterceptors.get();
         return () -> {
-            asyncInterceptors.forEach(AsyncInvocationInterceptor::applyContext);
+            if (asyncInvocationInterceptors != null) {
+                asyncInvocationInterceptors.forEach(AsyncInvocationInterceptor::applyContext);
```

Comment:
[Javadoc for the factory](https://github.com/eclipse/microprofile-rest-client/blob/master/api/src/main/java/org/eclipse/microprofile/rest/client/ext/AsyncInvocationInterceptorFactory.java#L31) says, that applyContext is called in order of descending priorities, so you might want to iterate in reverse order in the wrappers

Reply:
Thank you. Now it should be in proper order.